src/views/skuLookup.tsx

```diff
@@ -0,0 +1,349 @@
+        inv.availabilityByProducts?.[0]?.availabilityDetails?.atp?.toString() ||
+        "",
+    };
+    let firstEmptyCountry = yantriksInventory.findIndex((inventory) => {
+      return inventory[SKU_INVENTORY.COUNTRY] == "";
+    });
+    yantriksInventory[firstEmptyCountry] = inventory;
+    setYantriksInventory([...yantriksInventory]);
+  };
+
+  const fetchSKUDetails = async () => {
+    setIsLoading(true);
+    try {
+      const catalogURL = SKU_LOOKUP_CATALOG.replace(":skuid", skuid);
+      const catalogResponse = await fetchData(catalogURL, {});
+      setIsLoading(false);
+      catalogResponse?.inventory?.forEach((item) => {
+        fetchInventoryStatus(item.country);
+      });
+      setStatesData(catalogResponse);
+    } catch (error) {
+      setIsLoading(false);
+      console.log("error fetching data ", error);
+    }
+  };
+  const fetchInventoryStatus = async (country) => {
+    setIsLoading(true);
+    try {
+      const inventoryURL = SKU_LOOKUP_INVENTORY_STATUS.replace(
+        ":country",
+        country
+      ).replace(":skuid", skuid);
+      const inventoryResponse = await fetchData(inventoryURL, {});
+      setIsLoading(false);
+      setYantriksData(inventoryResponse, country);
+    } catch (error) {
+      setIsLoading(false);
+      console.log("error fetching data ", error);
+    }
+  };
+
+  const onSearch = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter" && skuid?.trim().length > 0) {
+      event.preventDefault();
+      fetchSKUDetails();
+    }
+  };
+
+  return (
+    <div className="sku-lookup h-full pb-4">
+      <div className="flex justify-center items-center gap-2">
+        <SearchBar
+          searchValue={skuid}
+          setSearchValue={setSkuId}
+          onSearch={onSearch}
+          placeholder="Search SKU#"
+          className="text-white-500"
+        />
+      </div>
+      {isLoading ? (
+        <Loader className="h-full" />
+      ) : (
+        <>
+          {width < SCREEN_WIDTH.SM ? (
+            <div className="py-4">
+              <SkuCard title={SKU_DETAILS.TITLE} cardData={skuDetailsTable} />
+              <SkuCard
+                title={SKU_PRICE_DETAILS.TITLE}
+                cardData={priceDetailsTable}
+                wrapperClass="grid-cols-[repeat(auto-fill,minmax(70px,1fr))]"
+              />
+              <SkuCard
+                title={SKU_INVENTORY.ATG_INVENTORY}
+                cardData={atgInventory}
+              />
+              <SkuCard
+                title={SKU_INVENTORY.YANTRIKS_INVENTORY}
+                cardData={yantriksInventory}
+              />
+              <SkuCard title={SKU_PRODUCT_INFO.TITLE} cardData={productInfo} />
+            </div>
+          ) : (
+            <>
+              <HorizontalTable
+                tableData={skuDetailsTable}
+                tableHead={SKU_DETAILS.TITLE}
+                rowHeadClassName="p-4 text-right font-medium w-[15%] lg:whitespace-nowrap"
```

Comment:
Try this approach to avoid repetitive invoking of the same component with minor differences.
Untested. So, test the changes.
```
[
[SKU_DETAILS.TITLE,  skuDetailsTable],
[SKU_PRICE_DETAILS.TITLE,  priceDetailsTable],
]
.map(e) => (
  <HorizontalTable
                tableData={e[0]}
                tableHead={e[1]}
                rowHeadClassName="p-4 text-right font-medium w-[12.5%] lg:whitespace-nowrap"
                dataClassName="text-left p-4 border-r border-black-400 w-[12.5%]"
              />
);
```

Reply:
Same for `<SkuCard/>` invocation also.